### PR TITLE
Bug 2120268: rook: add logCollector in storageCluster

### PR DESCRIPTION
**Pull request by subhamkrai**
rook: upgrade rook to v1.9.9

rook: add logCollector in storageCluster

adding logCollector in storageCluster which will have
fields like enabled to enable/disable logCollector,
periodicity interval after which logs will be rotated
and maxLogSize time after which logs will be rotated.
If both periodicity and maxLogSize what every comes
first will rotate the logs.

### CODE REVIEW

**Comment by subhamkrai** [comment]
/assign @travisn 
seems like BP to 4.10 will be difficult 
```
/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
vendor/github.com/hashicorp/vault/sdk/helper/pluginutil/run_config.go:13:2: //go:build comment without // +build comment
Error: not all generators ran successfully
```

we are upgrading rook from 1.8.3 to 1.9.9 there could be many errors similar to this.

cc @agarwal-mudit @iamniting

**Comment by travisn** [comment]
> /assign @travisn seems like BP to 4.10 will be difficult
> 
> ```
> /go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
> vendor/github.com/hashicorp/vault/sdk/helper/pluginutil/run_config.go:13:2: //go:build comment without // +build comment
> Error: not all generators ran successfully
> ```
> 
> we are upgrading rook from 1.8.3 to 1.9.9 there could be many errors similar to this.
> 
> cc @agarwal-mudit @iamniting

@subhamkrai Instead of backporting the PR from 4.11 that has all the changes to the go modules, let's create a PR with only the log rotate setting and regenerate the go modules based on the 4.10 branch. Then the change will be much smaller and the 4.11 go module updates won't cause backport issues.

File: go.mod [hunk 1]
```diff
@@ -28,25 +25,24 @@ require (
 	github.com/operator-framework/operator-lib v0.6.0
 	github.com/operator-framework/operator-lifecycle-manager v0.17.0
 	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.46.0
-	github.com/prometheus/client_golang v1.11.0
+	github.com/prometheus/client_golang v1.11.1
 	github.com/prometheus/client_model v0.2.0
-	github.com/rook/rook v1.8.3
+	github.com/rook/rook v1.9.9
```
Comment (travisn) [inline go.mod:30]: The 4.10 branch must be based on v1.8, not on v1.9

Comment (subhamkrai) [reply go.mod:30]: so how we'll get API from rook based on 1.9.9?

**Review by subhamkrai — COMMENTED** [review]
this changes are coming `make gen-latest-csv` and `make update-generated` https://github.com/red-hat-storage/ocs-operator/pull/1788/commits/a013e88aa649f194add321e61664a1472b597181

File: deploy/bundle/manifests/cephblockpool.crd.yaml [hunk 2]
```diff
@@ -14,7 +14,11 @@ spec:
     singular: cephblockpool
   scope: Namespaced
   versions:
-  - name: v1
+  - additionalPrinterColumns:
```
Comment (travisn) [inline deploy/bundle/manifests/cephblockpool.crd.yaml:17]: Changes aren't expected in this crd

File: deploy/bundle/manifests/cephblockpoolradosnamespace.crd.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+---
```
Comment (travisn) [inline deploy/bundle/manifests/cephblockpoolradosnamespace.crd.yaml:1]: not expected to backport this crd, let's see if we can restrict to only the changes needed for the log collector

File: deploy/bundle/manifests/cephfilesystem.crd.yaml [hunk 4]
```diff
@@ -314,6 +314,89 @@ spec:
                     nullable: true
                     type: object
                     x-kubernetes-preserve-unknown-fields: true
+                  livenessProbe:
```
Comment (travisn) [inline deploy/bundle/manifests/cephfilesystem.crd.yaml:317]: not expected

File: hack/common.sh [hunk 5]
```diff
@@ -44,9 +44,9 @@ NOOBAA_CSV="$OUTDIR_TEMPLATES/noobaa-csv.yaml"
 ROOK_CSV="$OUTDIR_TEMPLATES/rook-csv.yaml.in"
 OCS_CSV="$OUTDIR_TEMPLATES/ocs-operator.csv.yaml.in"
 
-LATEST_ROOK_IMAGE="rook/ceph:v1.8.3"
-LATEST_NOOBAA_IMAGE="noobaa/noobaa-operator:master-20210912"
-LATEST_NOOBAA_CORE_IMAGE="noobaa/noobaa-core:master-20210912"
+LATEST_ROOK_IMAGE="rook/ceph:v1.9.9"
```
Comment (travisn) [inline hack/common.sh:47]: We really don't expect to use a 1.9.9 image in the 4.10 branch, which is based on 1.8

**Review by travisn — CHANGES_REQUESTED** [review]

File: hack/common.sh (hunk 5 above)
Comment (subhamkrai) [reply hack/common.sh]: reverted, we should be 1.8 based only now

File: api/v1/storagecluster_types.go [hunk 6]
```diff
@@ -79,6 +79,11 @@ type StorageClusterSpec struct {
 	// AllowRemoteStorageConsumers Indicates that the OCS cluster should deploy the needed
 	// components to enable connections from remote consumers.
 	AllowRemoteStorageConsumers bool `json:"allowRemoteStorageConsumers,omitempty"`
+
+	// Logging represents loggings settings
```
Comment (umangachapagain) [inline api/v1/storagecluster_types.go:83]:
```suggestion
	// LogCollector represents log collector settings
```

**Review by umangachapagain — CHANGES_REQUESTED** [review]

**Comment by umangachapagain** [comment]
Can't find the logs for tests. Please test it manually once.

**Comment by subhamkrai** [comment]
> Can't find the logs for tests. Please test it manually once.

I have tested the original pr and also verified by qe(IIRC) it is just backport to 4.10

File: hack/common.sh [hunk 7]
```diff
@@ -44,7 +44,7 @@ NOOBAA_CSV="$OUTDIR_TEMPLATES/noobaa-csv.yaml"
 ROOK_CSV="$OUTDIR_TEMPLATES/rook-csv.yaml.in"
 OCS_CSV="$OUTDIR_TEMPLATES/ocs-operator.csv.yaml.in"
 
-LATEST_ROOK_IMAGE="rook/ceph:v1.8.3"
+LATEST_ROOK_IMAGE="subham03/rook:down-4.10"
```
Comment (subhamkrai) [inline hack/common.sh:47]:
```suggestion
LATEST_ROOK_IMAGE="subham03/rook:down-4.10"
```
will change this later, before pr merges

File: api/v1/storagecluster_types.go (hunk 6 above)
Comment (subhamkrai) [reply api/v1/storagecluster_types.go]: done

**Pull request merged**